### ISSUE TITLE
nearbyDroids sample working?

### ISSUE THREAD

**Issue by danbatis** [playgameservices/play-games-plugin-for-unity#2677]
I have just built the nearby sample, the nearbyDroids project, it seems to be working but I deployed to 2 identical Samsung galaxy 7 and neither can find the other when attempting to multiplay. Also, the apps never asked any permission related to bluetooth, location, wifi or anything really. I was expecting it to ask for the nearby related permissions.
Built with Unity 2018.4.5f1 in Mac OS 10.12.6
Created a new project, imported the googleplaygames package, the nearbydroids package, switched to android, did the setup nearby and set the name of the nearby connection service. I was having a popup complaining about the google play services, so I created a fake resources template like:
<?xml version="1.0" encoding="utf-8"?>

<resources>
 <string name="app_id">123456</string>
 <string name="package_name">com.Semel.NearbyRaw</string>
</resources>

And then run the android setup from the menu google play games setup.
Now I get zero errors, but the devices still do not recognize each other.
Any clues?

**Comment by olehkuznetsov** [comment]
Yes, you are right, and it should ask for permissions. We will update the sample.

**Comment by ozdemir08** [comment]
I took a look at it. You should see in logs that it's giving error regarding "Missing Permissions". But even after adding permissions, it still has some errors after connection between devices are established. I'll look into it sometime, too.

But if you want to test nearby connections and multiplayer APIs, you can use SmokeTest and QuizRacer sample apps for now.

**Comment by eirikost** [comment]
Hey @ozdemir08, I tried to upgrade from v. 9.54 to 10.6 and I also get this problem although the android manifest is the same.

```
01-13 20:28:50.831  4715 10896 E NearbyConnections: Missing required permissions, aborting call to startDiscovery() for serviceId com.******* [CONTEXT service_id=54 ]
```

Sidetrack: I tried checking the source code but then I noticed that reliable and unreliable messages are no longer treated differently. I guess it is not possible anymore, but do you know why?

**Comment by ozdemir08** [comment]
NearbyConnection needs [ACCESS_COARSE_LOCATION](https://developer.android.com/reference/android/Manifest.permission.html#ACCESS_COARSE_LOCATION) and [ACCESS_FINE_LOCATION](https://developer.android.com/reference/android/Manifest.permission.html#ACCESS_FINE_LOCATION). Note that both are [dangerous permissions](https://developer.android.com/guide/topics/permissions/overview.html#dangerous_permissions). Even though you have a permission explicitly specified in AndroidManifest.xml, you should still ask the same permission when the app starts as in [here](https://docs.unity3d.com/Manual/android-RequestingPermissions.html). After the user gives the permission, then you are good to go. 

My understanding from what you wrote is that putting permissions into AndroidManifest.xml was enough for you to use NearbyAPIs with v9.54. I haven't tried it with v9.54 so I am not sure what the reason is. But my guess is that permissions being classified as dangerous or not was probably introduced after this plugin's v9.54 release. So, it is probably about the version of a library that this plugin depends on. 


> Sidetrack: I tried checking the source code but then I noticed that reliable and unreliable messages are no longer treated differently. I guess it is not possible anymore, but do you know why?

There had been some changes a few years ago in the logic of Nearby APIs and even though there were reliable and unreliable ways to send messages on the interface level, they were being handled the same way. With the v10, we rewrote the plugin without changing any interfaces. That's why we still have methods for reliable and unreliable messages with the only difference that them being handled in the same way is now visible in the code.

**Comment by eirikost** [comment]
Thanks a lot! The fine location permission was turned off for the app on one of the phones for some reason. Checking permissions in game was a nice feature which I have not used before. Found out that having "uses-feature" for fine location in the android manifest was not enough anymore but it worked with "uses-permission".

I get a lot of messages saying that some methods only can be called from the main thread, but they might not be related to GPGS. I think i can upgrade now 🙌 

````
01-15 21:42:11.122 11098 11098 E Unity   : UnityException: get_isDebugBuild can only be called from the main thread.
01-15 21:42:11.122 11098 11098 E Unity   : Constructors and field initializers will be executed from the loading thread when loading a scene.
01-15 21:42:11.122 11098 11098 E Unity   : Don't use this function in the constructor or field initializers, instead move initialization code to the Awake or Start function.
01-15 21:42:11.122 11098 11098 E Unity   :   at MultiplayerController.Log (System.String text) [0x00000] in <00000000000000000000000000000000>:0
01-15 21:42:11.122 11098 11098 E Unity   :   at MultiplayerController.ShowMPStatus (System.String message) [0x00000] in <00000000000000000000000000000000>:0
01-15 21:42:11.122 11098 11098 E Unity   :   at MultiplayerController.StartNearbyGame () [0x00000] in <00000000000000000000000000000000>:0
01-15 21:42:11.122 11098 11098 E Unity   :   at MultiplayerController+<>c__DisplayClass64_0.<SendConnectionRequest>b__0 (GooglePlayGames.BasicApi.Nearby.ConnectionResponse response) [0x00000] in <00000000000000000000000000000000>:0
01-15 21:42:11.122 11098 11098 E Unity   :   at System.Action`1[T].Invoke (T obj) [0x00000] in <00000000000000000000000000000000>:0
01-15 21:42:11.122 11098 11098 E Unity   :   at GooglePlayGames.Android.AndroidNearbyConnectionClient+DiscoveringConnectionLifecycleCallback.onConnectionResult (System

01-15 22:30:03.105 26256 26256 E Unity   : Function void SoundChannelInstance::UpdatePauseState() may only be called from main thread!
01-15 22:30:03.105 26256 26256 E Unity   : Menu:StartGame(Int32)
01-15 22:30:03.105 26256 26256 E Unity   : MultiplayerController:StartNearbyGame()
01-15 22:30:03.105 26256 26256 E Unity   : <>c__DisplayClass64_0:<SendConnectionRequest>b__0(ConnectionResponse)
01-15 22:30:03.105 26256 26256 E Unity   : System.Action`1:Invoke(T)
01-15 22:30:03.105 26256 26256 E Unity   : GooglePlayGames.Android.DiscoveringConnectionLifecycleCallback:onConnectionResult(String, AndroidJavaObject)
01-15 22:30:03.105 26256 26256 E Unity   : System.Reflection.MonoMethod:Invoke(Object, BindingFlags, Binder, Object[], CultureInfo)
01-15 22:30:03.105 26256 26256 E Unity   : UnityEngine.AndroidJavaProxy:Invoke(String, Object[])
01-15 22:30:03.105 26256 26256 E Unity   : UnityEngine._AndroidJNIHelper:InvokeJavaProxyMethod(AndroidJavaProxy, IntPtr, IntPtr)
01-15 22:30:03.105 26256 26256 E Unity   :
01-15 22:30:03.105 26256 26256 E Unity   : [./Modules/Audio/Public/sound/SoundManager.cpp line 35]
01-15 22:30:03.105 26256 26256 E Unity   : (Filename: ./Modules/Audio/Public/sound/SoundManager.cpp Line: 35)
````

**Comment by Thomas-6freedom** [comment]
I succeeded to connect two devices after checking the permissions manually.
But when joining, I can't see the other player in the player List on the right.

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
